test/integration/integration_test.cc

```diff
@@ -1918,6 +1918,14 @@ TEST_P(IntegrationTest, Preconnect) {
     clients.front()->close();
     clients.pop_front();
   }
+
+  for (auto& connection : fake_connections) {
+    AssertionResult result = connection->close();
```

Comment:
what do you think of just doing 
ASSERT_TRUE(connection->close());
ASSERT_TRUE(connection->waitForDisconnect());
connection.reset()
?
I think it's more consistent with the rest of the code and the test will fail and make clear where it failed either way I think?

Reply:
Yes, totally. This is a bad cut/paste.